src/SourceGen/src/Core/AppBuilderGenerator.cs

```diff
@@ -0,0 +1,365 @@
+					{
+						HasAppDelegateSubclass = true;
+					}
+				}
+			}
+		}
+
+		class AndroidMauiBuilderReceiver
+		{
+			public bool HasApplicationSubclass { get; set; } = false;
+			public TypeInfoParts? PartialApplicationSubclass { get; set; } = null;
+			public bool HasPartialApplicationCreateOverride { get; set; } = false;
+
+			public bool HasMainLauncher { get; set; } = false;
+
+			public void OnVisitSyntaxNode(GeneratorSyntaxContext context, SyntaxNode syntaxNode)
+			{
+				if (!HasMainLauncher && syntaxNode is AttributeSyntax attrSyntax)
+				{
+					var attrFullName = attrSyntax.Name.ToFullString();
+					if (attrSyntax.Name.ToFullString() == "global::Android.App.Activity")
+					{
+						HasMainLauncher = attrSyntax.ArgumentList != null
+								&& attrSyntax.ArgumentList.ChildNodes().Any(cn => cn is AttributeArgumentSyntax aas
+									&& aas.NameEquals?.Name?.Identifier.ValueText == "MainLauncher"
+									&& aas.Expression is ExpressionSyntax exs
+									&& exs.Kind() == SyntaxKind.TrueLiteralExpression);
+					}
+				}
+
+				// Look for an AppDelegate implementation of some sort
+				if (syntaxNode is SimpleBaseTypeSyntax baseTypeSyntax)
+				{
+					var baseId = GlobalizeNs(baseTypeSyntax.ToFullString().Trim());
+
+					if (baseId.Equals("global::Microsoft.Maui.MauiApplication") 
+						|| baseId.Equals("global::Android.App.Application"))
+					{
+						HasApplicationSubclass = true;
+
+						if (baseId.Equals("global::Microsoft.Maui.MauiApplication"))
+						{
+							// Check if it's a partial declaration, if so, get the type name back and see if it's partial to us
+							// or if it's something else
+							var partialClassSyntax = baseTypeSyntax.Ancestors().FirstOrDefault(n =>
+								n is ClassDeclarationSyntax cp
+								&& cp.Modifiers.Any(m => m.IsKind(SyntaxKind.PartialKeyword)));
+
+							if (partialClassSyntax is ClassDeclarationSyntax classDeclParent)
+							{
+								PartialApplicationSubclass = classDeclParent.GetTypeInfoParts();
+
+								// Now check if they override CreateMauiApp() to see if we need to emit it or not
+								HasPartialApplicationCreateOverride = HasCreateMauiAppOverride(context, classDeclParent);
+							}
+						}
+					}
+				}
+			}
+		}
+
+		class AppStartupSyntaxReceiver : ISyntaxContextReceiver
+		{
+			public readonly MaciOSMauiAppBuilderReceiver MaciOS = new ();
+			public readonly AndroidMauiBuilderReceiver Android = new ();
+
+			public string? MauiAppBuilderMethod { get; set; } = null;
+
+			public void OnVisitSyntaxNode(GeneratorSyntaxContext context)
+			{
+				var syntaxNode = context.Node;
+
+				// Look for the static method entry point that returns a MauiApp
+				if (syntaxNode is MethodDeclarationSyntax mds)
+				{
+					if (mds.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)))
+					{
+						var returnTypeFullNamme = context.GetReturnType(mds);
+
+						if (returnTypeFullNamme == "Microsoft.Maui.MauiApp")
```

Comment:
maybe we can rely on symbols instead of strings, to make it safer (?)

Reply:
Yeah. I totally forgot my usual "this should at the very least be a const and not magic string littered about". 😄